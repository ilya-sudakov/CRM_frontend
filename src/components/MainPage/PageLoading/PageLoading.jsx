import React from 'react';
import './PageLoading.scss';

const PageLoading = () => {
  return (
    <div className="page-loading">
      <div className="page-loading__wrapper">
        <div className="page-loading__circle"></div>
      </div>
    </div>
  );
};

export default PageLoading;
