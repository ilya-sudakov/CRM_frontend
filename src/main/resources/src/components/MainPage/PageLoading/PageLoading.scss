@import "../../../variables.scss";

.page_loading {
  width: 100%;
  // height: 100%;
  padding: 25px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-top: 25vh;

  &__title {
    font-size: 200%;
  }

  .lds-ellipsis {
    display: inline-block;
    position: relative;
    width: 80px;
    height: 80px;
    margin-top: -15px;
  }
  .lds-ellipsis div {
    position: absolute;
    top: 33px;
    width: 13px;
    height: 13px;
    border-radius: 50%;
    background: $button__background--inverted;
    animation-timing-function: cubic-bezier(0, 1, 1, 0);
  }
  .lds-ellipsis div:nth-child(1) {
    left: 8px;
    animation: lds-ellipsis1 0.6s infinite;
  }
  .lds-ellipsis div:nth-child(2) {
    left: 8px;
    animation: lds-ellipsis2 0.6s infinite;
  }
  .lds-ellipsis div:nth-child(3) {
    left: 32px;
    animation: lds-ellipsis2 0.6s infinite;
  }
  .lds-ellipsis div:nth-child(4) {
    left: 56px;
    animation: lds-ellipsis3 0.6s infinite;
  }
  @keyframes lds-ellipsis1 {
    0% {
      transform: scale(0);
    }
    100% {
      transform: scale(1);
    }
  }
  @keyframes lds-ellipsis3 {
    0% {
      transform: scale(1);
    }
    100% {
      transform: scale(0);
    }
  }
  @keyframes lds-ellipsis2 {
    0% {
      transform: translate(0, 0);
    }
    100% {
      transform: translate(24px, 0);
    }
  }
}
