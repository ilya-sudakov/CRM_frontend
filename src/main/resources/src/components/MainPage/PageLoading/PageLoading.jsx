import React from 'react';
import './PageLoading.scss';

const PageLoading = () => {
    return (
        <div className="page_loading">
            <div className="page_loading__title">Идет загрузка...</div>
            <div class="lds-ellipsis">
                <div></div><div></div><div></div><div></div>
            </div>
        </div>
    )
}

export default PageLoading;