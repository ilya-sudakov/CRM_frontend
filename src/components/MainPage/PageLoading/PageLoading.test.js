import React from 'react';
import PageLoading from './PageLoading.jsx';
import { render, cleanup } from '@testing-library/react';

describe('PageLoading component', () => {
  afterEach(cleanup);

  it('matches snapshot', () => {
    const { asFragment } = render(<PageLoading />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders', () => {
    render(<PageLoading />);
  });
});
