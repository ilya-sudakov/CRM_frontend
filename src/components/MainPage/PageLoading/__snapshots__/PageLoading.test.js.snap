// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoading component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-loading"
  >
    <div
      class="page-loading__wrapper"
    >
      <div
        class="page-loading__circle"
      />
    </div>
  </div>
</DocumentFragment>
`;
